src/main/java/bg/ehealth/prescriptions/config/gateways/BdaGatewayProperties.java

```diff
@@ -0,0 +1,50 @@
+package bg.ehealth.prescriptions.config.gateways;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.ConstructorBinding;
+import org.springframework.boot.context.properties.bind.DefaultValue;
+import org.springframework.boot.convert.DurationUnit;
+import org.springframework.validation.annotation.Validated;
+
+import javax.validation.constraints.NotBlank;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+
+@ConfigurationProperties("bda.gateway")
+@ConstructorBinding
+@Validated
+public class BdaGatewayProperties {
+
+    @NotBlank
+    private final String host;
+    @NotBlank
+    private final String registersUrl;
+    @NotBlank
+    private final String registerPharmacies;
+    @DurationUnit(ChronoUnit.SECONDS)
+    private final Duration connectionTimeout;
+
+    public BdaGatewayProperties(String host, String registersUrl, String registerPharmacies,
```

Comment:
let's avoid constructor injection (e.g. it makes it difficult to use mocks).
I think plain old @Value("${..}" annotations (or if there's a non-constructor option for such binding).
Another good aspect of @Value is that in the future people will be able to find the properies by just searching through the project.